ddtrace/_trace/span.py

```diff
@@ -595,6 +595,22 @@ def context(self) -> Context:
             self._context = Context(trace_id=self.trace_id, span_id=self.span_id, is_remote=False)
         return self._context
 
+    @property
+    # This can't actually return None, but we say it can to satisfy type checking.
+    def _local_root(self) -> Optional["Span"]:
+        if self._local_root_value is None:
+            return self
+        return self._local_root_value
+
+    @_local_root.setter
+    def _local_root(self, value: "Span") -> None:
+        if value is not self:
```

Comment:
do we need an `else` branch here? like, what if we have a `self._local_root_value` set already, but then we try to update with `span._local_root = span`... do we then need to do `else: self._local_root_value = None` ?

I know this case is highly unlikely, but given this code it is technically possible.

Reply:
For what it's worth, I thought about this and only didn't include it because I figured there would never be a scenario in which a child span could turn into a root span. Seems totally fine to add to be defensive.

Reply:
I think from the code, we almost never set this, other than on initialization anyways?

Part of the reason I like the parent linked list traversal approach.. definitely higher overhead on reads, but probably simplifies a bunch of logic/code in places.

"like" - no strong opinion, no need to actually change that. this approach does work and covers every case Erika and I walked through.

Reply:
Yeah, it's not set outside of initialization.

The parent traverse removes 1 line from tracer.py and changes the _local_root to a recursive function. Doesn't seem like much of a simplification?

On the flip-side, you introduce a limit of 1000 nested spans because of the recursive function call stack limit in Python. That could be fine in practice, but seems like an unnecessary restriction and worsening of performance, no?